browser/brave_news/brave_news_tab_helper.cc

```diff
@@ -166,13 +173,32 @@ void BraveNewsTabHelper::OnReceivedRssUrls(const GURL& site_url,
     return;
   }
 
-  // First things first, we just store the urls. This let's us know that we have
-  // feeds, so we should show the button.
-  for (const auto& url : feed_urls) {
-    rss_page_feeds_.push_back({.feed_url = url, .title = ""});
+  // For each feed, we need to check if its a combined feed.
+  for (const auto& feed_url : feed_urls) {
+    controller_->GetPublisherForFeed(
+        feed_url,
+        base::BindOnce(
+            [](base::WeakPtr<BraveNewsTabHelper> tab_helper, GURL feed_url,
+               brave_news::mojom::PublisherPtr publisher) {
+              if (!tab_helper) {
+                return;
+              }
+
+              FeedDetails feed;
```

Comment:
Could it be possible that the tab url has been changed? If then I think we should compare tab url with feed_url and return. (maybe other places)

Reply:
`OnPrimaryPageChanged` calls `weak_ptr_factory_.InvalidateWeakPtrs()` so I think this should be okay

Reply:
Aha! I see. Thanks!